repository_service_tuf_worker/signer.py

```diff
@@ -1,22 +1,83 @@
 # SPDX-FileCopyrightText: 2024 Repository Service for TUF Contributors
 #
 # SPDX-License-Identifier: MIT
+import os
+from pathlib import Path
+from typing import Optional
 
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
 from dynaconf import Dynaconf
 from securesystemslib.signer import (
     SIGNER_FOR_URI_SCHEME,
     CryptoSigner,
     Key,
+    SecretsHandler,
     Signer,
 )
 
 from repository_service_tuf_worker.interfaces import IKeyVault
 
+
+class FileNameSigner(CryptoSigner):
+    """File-based signer implementation.
+
+    Provide method to load **unencrypted** PKCS8/PEM private key from file.
+
+    File path is constructed by joining base path in envrionment variable
+    ``RSTUF_ONLINE_KEY_DIR`` with file in ``priv_key_uri``.
+
+    NOTE: Make sure to use the secrets management service of your deployment
+    platform to protect your private key!
+
+    Example::
+
+        RSTUF_ONLINE_KEY_DIR (env) "/run/secrets"
+        priv_key_uri (arg): "fn:foo"
+
+        File path: "/run/secrets/foo"
+
+    Raises:
+        KeyError: RSTUF_ONLINE_KEY_DIR environment variable not set
+        OSError: file cannot be loaded
+        ValueError: uri has no filenmae, or private key cannot be decoded,
+                or type does not match public key
+        cryptography.exceptions.UnsupportedAlgorithm: key type not supported
+
+    """
+
+    SCHEME = "fn"
+    DIR_VAR = "RSTUF_ONLINE_KEY_DIR"
```

Comment:
I think the settings (env variable) can be more specific as it is not for all online key schemas, only for  `fn`
`RSTUF_ONLINE_KEY_FN_DIR`.

Reply:
I'm fine either way, although I think that `RSTUF_ONLINE_KEY_DIR` seems pretty self-explanatory - directory for rstuf online key - and adding an `fn` infix doesn't make it clearer but more confusing.

I assume you want to bind the name to the uri scheme or to FileNameSigner? Ideally, the user doesn't need to know about that binding (I'll talk more about it below).